include/deal.II/matrix_free/matrix_free.h

```diff
@@ -225,7 +225,8 @@ class MatrixFree : public Subscriptor
       const bool         initialize_mapping  = true,
       const bool         overlap_communication_computation    = true,
       const bool         hold_all_faces_to_owned_cells        = false,
-      const bool         cell_vectorization_categories_strict = false)
+      const bool         cell_vectorization_categories_strict = false,
+      const bool         pedantic_regarding_ghosted_values    = false)
```

Comment:
I think this makes sense. But I would use a different name for the variable, maybe
```suggestion
      const bool         allow_ghosted_vectors_in_loops    = true)
```
What do you think? (I think 'allow' is the right term here because throwing an assertion, even though it has no other effect, is like disallowing it.)

Reply:
Done!